pandas/tests/groupby/test_groupby.py

```diff
@@ -876,14 +871,45 @@ def test_omit_nuisance(df):
         grouped.agg(lambda x: x.sum(0, numeric_only=False))
 
 
-def test_omit_nuisance_sem(df):
-    # GH 38774 - sem should work with nuisance columns
+@pytest.mark.parametrize(
+    "agg_function",
+    ["max", "min"],
+)
+def test_keep_nuisance_agg(df, agg_function):
+    # GH 38815
     grouped = df.groupby("A")
-    result = grouped.sem()
-    expected = df.loc[:, ["A", "C", "D"]].groupby("A").sem()
+    result = getattr(grouped, agg_function)().columns
+    expected = df.loc[:, ["B", "C", "D"]].columns
+    np.testing.assert_array_equal(result, expected)
```

Comment:
*never* use `np.testing`, in this case use `tm.assert_frame_equal` (don'tjust compare the columns)

 I think we have a rule which should fail tests about this cc @MarcoGorelli

Reply:
Yeah I thought there was one - currently on holiday, will check next week when I'm back

Reply:
@horaceklai if you can change this one as well